pkg/security/scc/byrestrictions.go

```diff
@@ -15,37 +15,50 @@ func (s ByRestrictions) Less(i, j int) bool {
 	return pointValue(s[i]) < pointValue(s[j])
 }
 
+// The following constants define the weight of the restrictions and used for
+// calculating the weight of the particular SCC. The lower the number, the more
+// restrictive SCC is. Make sure that weak restrictions are always valued
+// higher than the combination of the strong restrictions.
+
+const privilegedWeight int = 20
+
+const hostVolumeWeight int = 10
+const nonTrivialVolumeWeight int = 5
+
+const runAsAnyUserWeight int = 4
+const runAsNonRootWeight int = 3
+const runAsRangeWeight int = 2
+const runAsUserWeight int = 1
+
 // pointValue places a value on the SCC based on the settings of the SCC that can be used
 // to determine how restrictive it is.  The lower the number, the more restrictive it is.
 func pointValue(constraint *securityapi.SecurityContextConstraints) int {
 	points := 0
 
-	// make sure these are always valued higher than the combination of the highest strategies
 	if constraint.AllowPrivilegedContainer {
-		points += 20
+		points += privilegedWeight
 	}
 
 	// add points based on volume requests
 	points += volumePointValue(constraint)
 
-	// strategies in order of least restrictive to most restrictive
-	switch constraint.SELinuxContext.Type {
-	case securityapi.SELinuxStrategyRunAsAny:
-		points += 4
-	case securityapi.SELinuxStrategyMustRunAs:
-		points += 1
+	// the map contains weights for both RunAsUser and SELinuxContext
+	// strategies by taking advantage that they have identical strategy names
+	strategiesWeights := map[string]int{
+		string(securityapi.RunAsUserStrategyRunAsAny):         runAsAnyUserWeight,
+		string(securityapi.RunAsUserStrategyMustRunAsNonRoot): runAsNonRootWeight,
+		string(securityapi.RunAsUserStrategyMustRunAsRange):   runAsRangeWeight,
+		string(securityapi.RunAsUserStrategyMustRunAs):        runAsUserWeight,
 	}
 
-	switch constraint.RunAsUser.Type {
-	case securityapi.RunAsUserStrategyRunAsAny:
-		points += 4
-	case securityapi.RunAsUserStrategyMustRunAsNonRoot:
-		points += 3
-	case securityapi.RunAsUserStrategyMustRunAsRange:
-		points += 2
-	case securityapi.RunAsUserStrategyMustRunAs:
-		points += 1
+	if weight, found := strategiesWeights[string(constraint.SELinuxContext.Type)]; found {
```

Comment:
discussed in https://github.com/openshift/origin/pull/15116#discussion_r127466583, we should at least log something.  An error is a backwards incompatible change and without a test that prevents us from shipping something that will throw the error I'd rather just log.